### PR TITLE
Travelling_Salesman.java

### LINKED ISSUE #675: Travelling Salesman Problem (TSP)

### PR DESCRIPTION
**Pull request by mansikagrawal**
Fixes #675 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request: Added Travelling_Salesman.java
- 
- 

#### Languages Used:Java
- 
- 

#### Files Added:Travelling_Salesman.java
- 
-

### CODE REVIEW

**Comment by somya-kapoor** [comment]
No indentation present in the code.
Also please try to edit in the existing PR.

**Comment by mansikagrawal** [comment]
@somya-kapoor  Please check mam now

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+import java.util.*;
+class TSP 
+{ 
+    // Function to find the minimum weight 
+    // Hamiltonian Cycle 
+    static int tspfn( int[][] graph, boolean[] visited, int currPosition, int n, int count, int cost, int answer) 
+     { 
+         // If last node is reached and it has a link 
+        // to the starting node i.e the source then
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: Please correct the indentation.

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    static int tspfn( int[][] graph, boolean[] visited, int currPosition, int n, int count, int cost, int answer) 
+     { 
+         // If last node is reached and it has a link 
+        // to the starting node i.e the source then 
+       // keep the minimum value out of the total cost 
+      // of traversal and "answer" 
+     // Finally return to check for more possible values 
+      if( count == n && graph[currPosition][0] > 0 ) 
+      { 
+        answer = Math.min( answer, cost + graph[currPosition][0]);
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: Please correct the indentation.

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      // of traversal and "answer" 
+     // Finally return to check for more possible values 
+      if( count == n && graph[currPosition][0] > 0 ) 
+      { 
+        answer = Math.min( answer, cost + graph[currPosition][0]); 
+        return answer; 	
+      } 
+    
+      // BACKTRACKING STEP 
+     // Loop to traverse the adjacency list
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: Please correct the indentation.

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        return answer; 	
+      } 
+    
+      // BACKTRACKING STEP 
+     // Loop to traverse the adjacency list 
+    // of currPos node and increasing the count 
+   // by 1 and cost by graph[currPos,i] value 
+   for( int i = 0; i < n; i++) 
+    { 
+      if( visited[i] == false && graph[currPosition][i] > 0)
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: Please correct the indentation.

**Comment by somya-kapoor** [comment]
@mansikagrawal Please correct the indentation.

**Comment by somya-kapoor** [comment]
@mansikagrawal Are you working on it?

**Comment by mansikagrawal** [comment]
@somya-kapoor Mam, please check

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+import java.util.*;
+class TSP 
+{ 
+    // Function to find the minimum weight 
+    // Hamiltonian Cycle 
+    static int tspfn( int[][] graph, boolean[] visited, int currPosition, int n, int count, int cost, int answer) 
+     { 
+        //If last node is reached and it has a link to the starting node i.e the source then 
+       //keep the minimum value out of the total cost of traversal and "answer"
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: inconsistency in indentation

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.*;
+class TSP 
+{ 
+    // Function to find the minimum weight 
+    // Hamiltonian Cycle 
+    static int tspfn( int[][] graph, boolean[] visited, int currPosition, int n, int count, int cost, int answer) 
+     { 
+        //If last node is reached and it has a link to the starting node i.e the source then 
+       //keep the minimum value out of the total cost of traversal and "answer" 
+      //Finally return to check for more possible values
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: inconsistency in indentation

File: Travelling_Salesman_Problem/Travelling_Salesman.java (hunk 2 above)
Comment (somya-kapoor) [reply Travelling_Salesman_Problem/Travelling_Salesman.java]: inconsistency in indentation

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        //If last node is reached and it has a link to the starting node i.e the source then 
+       //keep the minimum value out of the total cost of traversal and "answer" 
+      //Finally return to check for more possible values 
+      if( count == n && graph[currPosition][0] > 0 ) 
+      { 
+        answer = Math.min( answer, cost + graph[currPosition][0]); 
+        return answer; 	
+      } 
+    
+      //BACKTRACKING STEP
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: inconsistency in indentation

File: Travelling_Salesman_Problem/Travelling_Salesman.java (hunk 4 above)
Comment (somya-kapoor) [reply Travelling_Salesman_Problem/Travelling_Salesman.java]: inconsistency in indentation

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+       //n is the number of nodes i.e. V 
+       int n;
+       System.out.println(" Enter the number of villages: ");
+       Scanner in = new Scanner( System.in );
+       n = in.nextInt();
+       int graph[][] = new int[n][n];
+       System.out.println( "Enter the cost matrix " );
+       for( int i=0; i < n; i++)
+       {
+         System.out.println( "Enter the cost matrix for village " + i+1 );
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: inconsistency in the indentation , use 4 space indentation

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+         for( int j=0; j<n; j++)
+         graph[i][j] = in.nextInt();
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: inconsistency in indentation

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+       int n;
+       System.out.println(" Enter the number of villages: ");
+       Scanner in = new Scanner( System.in );
+       n = in.nextInt();
+       int graph[][] = new int[n][n];
+       System.out.println( "Enter the cost matrix " );
+       for( int i=0; i < n; i++)
+       {
+         System.out.println( "Enter the cost matrix for village " + i+1 );
+         for( int j=0; j<n; j++)
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: Space around operator

File: Travelling_Salesman_Problem/Travelling_Salesman.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+       boolean[] visited = new boolean[n]; 
+    
+      //Mark 0th node as visited 
+      visited[0] = true; 
+      int answer = Integer.MAX_VALUE; 
+    
+     //Find the minimum weight Hamiltonian Cycle 
+     answer = tspfn( graph, visited, 0, n, 1, 0, answer); 
+     //answer is the minimum weight Hamiltonian Cycle 
+     System.out.println(" The minimum cost is " +answer);
```
Comment (somya-kapoor) [inline Travelling_Salesman_Problem/Travelling_Salesman.java]: space around +

**Pull request closed without merging**